### PR TITLE
Implement note feature

### LINKED ISSUE #56: Add note feature
As a user, I can add personal health information for each contact, so that I can store additional information associated with the client.

Related issues:
* #57 - We need to display the note of each contact as well.

### PR DESCRIPTION
**Pull request by ruijietay**
Referencing #56, added the note feature.

This includes adding the relevant test case for the note feature, as well as the updating of the `edit` command such that we can update notes for each contact. We can now also use the `note` command to add a new note to a contact.

Closes #56.

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/NoteCommandTest.java
Comment (dillontkh) [inline src/test/java/seedu/address/logic/commands/NoteCommandTest.java:1]: Maybe we could add a test for removing a note too?

**Review by dillontkh — COMMENTED** [review]
Great work! The note feature works perfectly! Just a few small nits.

File: src/main/java/seedu/address/logic/commands/NoteCommand.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            + "Existing note will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NOTE + "NOTE] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "note/ Likes to swim.";
+
+    public static final String MESSAGE_NOT_IMPLEMENTED_YET =
+            "Note command not implemented yet";
+
+    public static final String MESSAGE_ARGUMENTS = "Index: %1$d, Note: %2$s";
```
[the remark this answers is not in the document]
Comment (ruijietay) [reply src/main/java/seedu/address/logic/commands/NoteCommand.java:35]: Good catch, I've deleted it

File: src/main/java/seedu/address/logic/commands/NoteCommand.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    public static final String COMMAND_WORD = "note";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the note of the person identified "
+            + "by the index number used in the last person listing. "
+            + "Existing note will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NOTE + "NOTE] "
+            + "Example: " + COMMAND_WORD + " 1 "
```
[the remark this answers is not in the document]
Comment (ruijietay) [reply src/main/java/seedu/address/logic/commands/NoteCommand.java:29]: Good suggestion! Edited

File: src/test/java/seedu/address/logic/commands/NoteCommandTest.java
Comment (ruijietay) [reply src/test/java/seedu/address/logic/commands/NoteCommandTest.java:1]: Good idea, will do!

File: src/main/java/seedu/address/model/person/Note.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Person's note in the address book.
+ * Guarantees: immutable; is always valid.
+ */
+public class Note {
+
+    public static final String MESSAGE_CONSTRAINTS = "notes can take any values, and it should not be blank";
```
[the remark this answers is not in the document]
Comment (ruijietay) [reply src/main/java/seedu/address/model/person/Note.java:11]: Good catch, resolved

**Comment by bgopi23** [comment]
good idea to make use of remark tutorial for this

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -85,10 +86,9 @@ public static Address parseAddress(String address) throws ParseException {
      * Parses a {@code String note} into a {@code Note}.
      * Leading and trailing whitespaces will be trimmed.
      */
```
Comment (dillontkh) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:88]: We should update the javadocs as this function no longer takes in a `String`
```suggestion
     * Parses a {@code Optional<String> note} into a {@code Note}.
     * Leading and trailing whitespaces will be trimmed.
     * If the {@code Optional} is empty, return a {@code Note} with an empty string.
     */
```

**Review by dillontkh — COMMENTED** [review]
Just one final change to JavaDocs required.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 4 above)
Comment (ruijietay) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Thanks for pointing that out!

**Review by bgopi23 — APPROVED** [review]
LGTM!

**Review by dillontkh — APPROVED** [review]
LGTM!

**Pull request merged**